app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/address/SubscribedChangeAddressController.scala

```diff
@@ -0,0 +1,95 @@
+import com.google.inject.Inject
+import play.api.mvc.{Call, MessagesControllerComponents, Result}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.config.{ErrorHandler, ViewConfig}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.controllers.SessionUpdates
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.controllers.actions.{AuthenticatedAction, RequestWithSessionData, SessionDataAction, WithAuthAndSessionDataAction}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.JourneyStatus.Subscribed
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.address.Address
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.{Error, JourneyStatus, SessionData}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.repos.SessionStore
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.services.{SubscriptionService, UKAddressLookupService}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.util.Logging
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.{controllers, views}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class SubscribedChangeAddressController @Inject()(
+  val errorHandler: ErrorHandler,
+  val ukAddressLookupService: UKAddressLookupService,
+  val sessionStore: SessionStore,
+  val authenticatedAction: AuthenticatedAction,
+  val sessionDataAction: SessionDataAction,
+  subscriptionService: SubscriptionService,
+  cc: MessagesControllerComponents,
+  val enterPostcodePage: views.html.address.enter_postcode,
+  val selectAddressPage: views.html.address.select_address,
+  val addressDisplay: views.html.components.address_display,
+  val enterUkAddressPage: views.html.address.enter_uk_address,
+  val enterNonUkAddressPage: views.html.address.enter_nonUk_address,
+  val isUkPage: views.html.address.isUk
+)(implicit val viewConfig: ViewConfig, val ec: ExecutionContext)
+    extends FrontendController(cc)
+    with Logging
+    with WithAuthAndSessionDataAction
+    with SessionUpdates
+    with AddressController[Subscribed] {
+
+  def validJourney(
+    request: RequestWithSessionData[_]
+  ): Either[Result, (SessionData, Subscribed)] =
+    request.sessionData.flatMap(s => s.journeyStatus.map(s -> _)) match {
+      case Some((sessionData, s: Subscribed)) => Right(sessionData -> s)
+      case _                                  => Left(Redirect(controllers.routes.StartController.start()))
+    }
+
+  def updateAddress(journey: Subscribed, address: Address)(
+    implicit hc: HeaderCarrier
+  ): EitherT[Future, Error, JourneyStatus] = {
+    val updatedSubscribedDetails = journey.subscribedDetails.copy(address = address)
+
+    if(journey.subscribedDetails === updatedSubscribedDetails){
+      EitherT.pure[Future,Error](journey)
+    } else {
+      subscriptionService
+        .updateSubscribedDetails(updatedSubscribedDetails)
+        .map(_ => journey.copy(subscribedDetails = updatedSubscribedDetails))
+    }
+  }
```

Comment:
what about the call to ES6? or is that separate ticket?

Reply:
there's a separate ticket for that